Node 15 upgrades npm to 7, which breaks prefix installs

When I set Node.js version `15.0.1` to be used, the `yarnSetup` task fails like so:

```
> Task :yarnSetup FAILED
npm ERR! code ENOENT
npm ERR! syscall lstat
npm ERR! path /home/boris/project/.gradle/yarn/yarn-latest/lib
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, lstat '/var/lib/hudson/jobs/latest-git-build/workspace/backend/.gradle/yarn/yarn-latest/lib'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
```

Using version `14.14.0` works fine. Any ideas?

Can confirm it's an issue with node 15:
```
$ .gradle/nodejs/node-v15.0.1-darwin-x64/bin/npm install --global --no-save --prefix hello yarn
npm ERR! code ENOENT
npm ERR! syscall lstat
npm ERR! path /Users/alanor/projects/test/hello/lib
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, lstat '/Users/alanor/projects/test/hello/lib'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/alanor/.npm/_logs/2020-10-26T09_45_13_687Z-debug.log```

If I create `hello/lib` then it works.

Honestly I think this is something `npm` should be handling (and evidently used to since this works in `npm` 6.14.8 but not 7.0.3) but it shouldn't be too hard for us to work around, we just need to create that folder

for me it is breaking even with `3.0.0-rc4`

```
npm ERR! code ENOENT
npm ERR! syscall lstat
npm ERR! path /Users/ntmetchev/Paxos/dev/metals/metals-web/node/yarn/yarn-v1.22.10/lib
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, lstat '/Users/ntmetchev/Paxos/dev/metals/metals-web/node/yarn/yarn-v1.22.10/lib'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/ntmetchev/.npm/_logs/2020-11-20T10_36_53_772Z-debug.log

> Task :metals-web:yarnSetup FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':metals-web:yarnSetup'.
> Process 'command '/Users/ntmetchev/Paxos/dev/metals/metals-web/node/node-v15.2.1-darwin-x64/bin/npm'' finished with non-zero exit value 254

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

```

(venv) ➜  temp gradle yarn 
Starting a Gradle Daemon, 1 incompatible and 1 stopped Daemons could not be reused, use --status for details

> Task :yarnSetup
npm ERR! code ENOENT
npm ERR! syscall lstat
npm ERR! path /Users/ntmetchev/Paxos/dev/temp/node/yarn/yarn-v1.22.10/lib
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, lstat '/Users/ntmetchev/Paxos/dev/temp/node/yarn/yarn-v1.22.10/lib'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/ntmetchev/.npm/_logs/2020-11-24T09_51_12_416Z-debug.log

> Task :yarnSetup FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':yarnSetup'.
> Process 'command '/Users/ntmetchev/Paxos/dev/temp/node/node-v15.2.1-darwin-x64/bin/npm'' finished with non-zero exit value 254

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 13s
2 actionable tasks: 2 executed
(venv) ➜  temp

--- a/src/main/kotlin/com/github/gradle/node/npm/task/NpmSetupTask.kt
+++ b/src/main/kotlin/com/github/gradle/node/npm/task/NpmSetupTask.kt
@@ -13,6 +13,7 @@ import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.TaskAction
 import org.gradle.kotlin.dsl.listProperty
+import java.io.File
 
 /**
  * npm install that only gets executed if gradle decides so.
@@ -65,7 +66,7 @@ open class NpmSetupTask : DefaultTask() {
         val version = nodeExtension.npmVersion.get()
         val directory = npmDir.get().asFile
         // npm < 7 creates the directory if it's missing, >= 7 fails if it's missing
-        directory.mkdirs()
+        File(directory, "lib").mkdirs()
         return listOf("install", "--global", "--no-save", "--prefix", directory.absolutePath,
                 "npm@$version") + args.get()
     }

--- a/src/main/kotlin/com/github/gradle/node/yarn/task/YarnSetupTask.kt
+++ b/src/main/kotlin/com/github/gradle/node/yarn/task/YarnSetupTask.kt
@@ -6,6 +6,7 @@ import com.github.gradle.node.variant.VariantComputer
 import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.OutputDirectory
+import java.io.File
 
 /**
  * Setup a specific version of Yarn to be used by the build.
@@ -33,7 +34,7 @@ open class YarnSetupTask : NpmSetupTask() {
         val yarnPackage = if (version.isNotBlank()) "yarn@$version" else "yarn"
         // npm < 7 creates the directory if it's missing, >= 7 fails if it's missing
         // create the directory since we use npm to install yarn.
-        yarnDir.asFile.mkdirs()
+        File(yarnDir.asFile, "lib").mkdirs()
         return listOf("install", "--global", "--no-save", "--prefix", yarnDir.asFile.absolutePath, yarnPackage)
                 .plus(args.get())
     }

--- a/src/test/groovy/com/github/gradle/node/npm/task/NpmInstall_integTest.groovy
+++ b/src/test/groovy/com/github/gradle/node/npm/task/NpmInstall_integTest.groovy
@@ -39,6 +39,28 @@ class NpmInstall_integTest extends AbstractIntegTest {
         result.task(":npmInstall").outcome == TaskOutcome.UP_TO_DATE
     }
 
+    def 'install packages with npm >= 7'() {
+        given:
+        writeBuild('''
+            plugins {
+                id 'com.github.node-gradle.node'
+            }
+            
+            node {
+                download = true
+                version = '15.2.1'
+                npmVersion = '7.0.1'
+            }
+        ''')
+        writeEmptyPackageJson()
+
+        when:
+        def result = build('npmInstall')
+
+        then:
+        result.task(":npmInstall").outcome == TaskOutcome.SUCCESS
+    }
+
     def 'install packages with npm and postinstall task requiring npm and node'() {
         given:
         writeBuild('''

--- a/src/test/groovy/com/github/gradle/node/yarn/task/YarnInstall_integTest.groovy
+++ b/src/test/groovy/com/github/gradle/node/yarn/task/YarnInstall_integTest.groovy
@@ -36,6 +36,41 @@ class YarnInstall_integTest extends AbstractIntegTest {
         result.outcome == TaskOutcome.UP_TO_DATE
     }
 
+    def 'install packages with yarn on npm >= 7'() {
+        given:
+        writeBuild('''
+            plugins {
+                id 'com.github.node-gradle.node'
+            }
+
+            node {
+                download = true
+                yarnWorkDir = file('build/yarn')
+                version = '15.2.1'
+                npmVersion = '7.0.1'
+            }
+        ''')
+        writeEmptyPackageJson()
+
+        when:
+        def result = buildTask('yarn')
+
+        then:
+        result.outcome == TaskOutcome.SUCCESS
+
+        when:
+        result = buildTask('yarn')
+
+        then:
+        result.outcome == TaskOutcome.SUCCESS
+
+        when:
+        result = buildTask('yarn')
+
+        then:
+        result.outcome == TaskOutcome.UP_TO_DATE
+    }
+
     def 'install packages with yarn and and postinstall task requiring node and yarn'() {
         given:
         writeBuild('''

